### PR TITLE
Don't leave cursors open

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Looks sane.
:rocket:

File: lib/private/legacy/OC_DB_StatementWrapper.php [hunk 1]
```diff
@@ -104,6 +104,17 @@ public function fetchOne($column = 0) {
...
 
+	/**
+	 * Closes the cursor, enabling the statement to be executed again.
+	 *
+	 * @deprecated Use Result::free() instead.
+	 *
+	 * @return bool TRUE on success or FALSE on failure.
+	 */
+	public function closeCursor() {
+		$this->statement->closeCursor();
```
Comment (kesselb) [inline lib/private/legacy/OC_DB_StatementWrapper.php:115]:
```suggestion
		return $this->statement->closeCursor();
```

File: lib/private/legacy/OC_DB_StatementWrapper.php [hunk 2]
```diff
@@ -104,6 +104,17 @@ public function fetchOne($column = 0) {
...
 	}
 
+	/**
+	 * Closes the cursor, enabling the statement to be executed again.
+	 *
+	 * @deprecated Use Result::free() instead.
+	 *
+	 * @return bool TRUE on success or FALSE on failure.
+	 */
+	public function closeCursor() {
```
Comment (kesselb) [inline lib/private/legacy/OC_DB_StatementWrapper.php:114]:
```suggestion
	public function closeCursor(): bool {
```

**Review by kesselb — APPROVED** [review]
Nice :+1:

File: lib/private/legacy/OC_DB_StatementWrapper.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/legacy/OC_DB_StatementWrapper.php:113]: fixed differently

**Comment by icewind1991** [comment]
closing the cursor before a return makes no functional difference and only clutters the code, the cursor will be closed automatically when the result goes out of scope.

it only makes sense to manually close cursors if there is significant work being done in a function after the results are fetched

**Comment by MorrisJobke** [comment]
> closing the cursor before a return makes no functional difference and only clutters the code, the cursor will be closed automatically when the result goes out of scope.

Was about to ask this as well. It looks really odd that we need to actively close the cursor there. Does this really fix the OCI cursor leak?

**Comment by nickvergessen** [comment]
Well at least the OC_DB methods leave cursors behind I don't know why and how, maybe because of the static-ness.

**Comment by nickvergessen** [comment]
And then I think it doesn't hurt and we should just make it a pattern in our code base. So when someone adds a second query or some code after your query they dont need to magically remember to add a closeCursor for the query above

**Review by MorrisJobke — APPROVED** [review]
Fine by me

**Pull request merged**